### PR TITLE
CLN: Remove raise if missing only controlling the error message

### PR DESCRIPTION
**Pull request by phofl**
- [x] xref #41170

The removed tests don't make sense anymore, since all is printed now but without a fixed order

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -1356,21 +1341,8 @@ def _validate_read_indexer(
...
-            not_found = key[missing_mask]
-
-            with option_context("display.max_seq_items", 10, "display.width", 80):
-                raise KeyError(
-                    "Passing list-likes to .loc or [] with any missing labels "
-                    "is no longer supported. "
-                    f"The following labels were missing: {not_found}. "
-                    "See https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"  # noqa:E501
-                )
+            not_found = list(set(key) - set(ax))
```
Comment (jreback) [inline pandas/core/indexing.py:1344]: we *may* wan to limit the display here to max_seq_items (e.g. if you have a huge list), but could investigate if this is actually an issue. it *can* be potentially if it takes time to render and for example we are NOT raising this to the user (that's not true in this case, speaking more generally here).

**Comment by jreback** [comment]
thanks @phofl

File: pandas/core/indexing.py (hunk 1 above)
Comment (phofl) [reply pandas/core/indexing.py:1344]: Yeah this is a good point. Will try to investigate, if we can do this when we supress the error.

**Comment by TomAugspurger** [comment]
Did this cause a performance regression?

https://pandas.pydata.org/speed/pandas/index.html#indexing.NumericSeriesIndexing.time_loc_slice?python=3.8&Cython=0.29.21&p-index_dtype=%3Cclass+%27pandas.core.indexes.numeric.UInt64Index%27%3E&p-index_structure=%27nonunique_monotonic_inc%27&commits=67c9385787c4d854b75a6f2c04fdf6886fdb1a06

**Comment by phofl** [comment]
This tick really looks similar to the one a few weeks earlier. But I will have a look.

Practically this should only impact cases where a KeyError is raised

**Comment by jreback** [comment]
@TomAugspurger if you wouldn't mind creating an issue about this regression

**Comment by TomAugspurger** [comment]
Yeah, looks like a false alarm. The latest benchmarks are back to normal.

**Comment by jreback** [comment]
oh ok great!

**Pull request merged**